### PR TITLE
Create "Recommended libraries" category page with subpages.

### LINKED ISSUE #427: Move all non-tooling libraries into a `Recommended libraries` page
### Which Section of the Documentation Needs Improving?

The pages on libraries we recommended for non-tooling purposes (e.g. GUIs, CLIs, etc.)

### What Can be Improved About This Section

Clearer organisation of recommended libraries

### How to Improve This Section

Rename `Utility libraries` to `Recommended libraries` (as suggested in https://github.com/UCL-ARC/python-tooling/issues/355#issuecomment-2069093754).

Move the GUI libraries page (added in #373) into `Recommended libraries`.

What other pages, if any should be moved? [Parallel and asychronous processing](https://github-pages.arc.ucl.ac.uk/python-tooling/pages/parallel-async.html)?

### PR DESCRIPTION
**Pull request by samcunliffe**
As I'm one the who was blocking #373 and thumbs down-ing all over the place, here's a proposal to resolve #427.

## Resolves
- #427 

## Follows on from
- #328 
- #373 

## tl;dr of changes
This PR:
* Adds a parent "Recommended libraries" page with children: guis, parallel and async.
* Split's the non-obvious "Utility libraries" into clis and logging and moves them also under the parent.
* Adds some blurb to the parent page itself, which should be scrutinised for clarity and readability please!

## Choices:
I decide "Recommended libraries" rather than "Recommended modules" or "Utlility <anything>". Because a) some things in my brain's definition of the tools/not tools split are Python modules but still tool-ish and b) modules is a Python name but libraries feels _relatively_ language agnostic (to me, at least). I'm thinking of newcomers to Python.

### CODE REVIEW

File: docs/pages/libraries/index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+layout: default
+has_children: true
+---
+
+# Recommended libraries
+
+Python has a large ecosystem of packages and modules beyond what is built in to Python.
+Here is a list of libraries and toolkits that we've used in our projects, and can recommend (or might recommend that you avoid!).
+
+The libraries listed here are not really "tooling" in the sense that some projects will need some of the functionalities here, but it is project-dependent, so we don't include any of these in [our template]().
```
Comment (samcunliffe) [inline docs/pages/libraries/index.md]:
```suggestion
The libraries listed here are not really "tooling" in the sense that some projects will need some of the functionalities here, but it is project-dependent, so we don't include any of these in [our template](https://github.com/UCL-ARC/python-tooling/blob/main/README.md#using-this-template).
```

🤦

**Review by p-j-smith — COMMENTED** [review]
nice! It makes things much clearer and easier to navigate. Based on the definition of non-tooling, perhaps some of these pages could also be moved into `Recommended libraries`?:

- https://github-pages.arc.ucl.ac.uk/python-tooling/pages/benchmarking-profiling.html
- https://github-pages.arc.ucl.ac.uk/python-tooling/pages/bindings.html
- https://github-pages.arc.ucl.ac.uk/python-tooling/pages/jupyter-notebooks.html
- https://github-pages.arc.ucl.ac.uk/python-tooling/pages/refactoring.html

**Comment by samcunliffe** [comment]
> nice! It makes things much clearer and easier to navigate. Based on the definition of non-tooling, perhaps some of these pages could also be moved into `Recommended libraries`?:
> 
>   * https://github-pages.arc.ucl.ac.uk/python-tooling/pages/benchmarking-profiling.html
> 
>   * https://github-pages.arc.ucl.ac.uk/python-tooling/pages/bindings.html
> 
>   * https://github-pages.arc.ucl.ac.uk/python-tooling/pages/jupyter-notebooks.html
> 
>   * https://github-pages.arc.ucl.ac.uk/python-tooling/pages/refactoring.html

I think: no | yes | no | no.

Although I'm not certain about the first no. For me, a profiler is a tool, not a library. Even though it looks like those are pip-nstallable and may be critical for projects. The jupyter notebooks page contains services to host notebooks and linter, which feel very tool-ish to me. Google Colab and `nbqa` definitely aren't libraries. Similarly for refactoring: sourcery.ai is a service and jedi is for IDEs and therefore... tool?

**Comment by dstansby** [comment]
If we're saying tooling is stuff *every* project will need (or at least should have), then I'd say. jupyter-notebooks -> libraries (I've worked on many packages that don't use notebooks).

Otherwise I agree with:

- benchmarking > stay in tooling
- bindings > move to libraries
- refactoring > stay in tooling

**Review by p-j-smith — APPROVED** [review]
very nice, much tidier
